addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/providers/OpenShiftServiceProvider.java

```diff
@@ -0,0 +1,51 @@
+package io.fabric8.launcher.osio.providers;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Produces;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.HttpHeaders;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.launcher.base.identity.IdentityVisitor;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.base.identity.UserPasswordIdentity;
+import io.fabric8.launcher.core.spi.Application;
+import io.fabric8.launcher.core.spi.IdentityProvider;
+import io.fabric8.launcher.osio.EnvironmentVariables;
+import io.fabric8.launcher.osio.openshift.OpenShiftService;
+import io.fabric8.launcher.osio.tenant.Tenant;
+
+import static io.fabric8.launcher.core.spi.Application.ApplicationType.OSIO;
+
+@RequestScoped
+public class OpenShiftServiceProvider {
+
+    @Produces
```

Comment:
@edewit: I saw this `@Produces` annotation without any mime-types in @gastaldi 's code as well, but why is it here? According to what I read in the docs this annotation a) needs an argument and b) it's to decide what to result to return to the browser/rest client that accesses it. But this is an internal service that will never return any information to any client, so why do we mark it with this annotation?

Reply:
I'm using `javax.enterprise.inject.Produces` here and the documentation that you refer to is `javax.ws.rs.Produces`. Used here it marks the method as a factory method that needs to be invoked by CDI to create the `OpenShiftService` for further injection

Reply:
Aaah! You're right of course, thx, I must have looked at one of the WS ones and assumed they were all the same. Wow, that's really confusing.